src/Engine/ProtoCore/Parser/AssociativeAST.cs

```diff
@@ -377,6 +377,11 @@ public override IEnumerable<Node> Children()
 
     public class ArrayNameNode : AssociativeNode
     {
+        internal virtual ArrayNameNode CreateInstance()
```

Comment:
why is this necessary - definitely missing something here, if these types already inherit from `ArrayNameNode` can't you just use `as` or cast?